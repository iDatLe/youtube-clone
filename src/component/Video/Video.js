import React from 'react';
import './Video.scss';

const BASE_EMBED_URL = 'https://www.youtube.com/embed/';

export function Video(props) {
    if (!props.id) {
        return null;
    }

    const embedUrl = `${BASE_EMBED_URL}${props.id}`;
    // if you want autoplay then add '?autoplay=1' at the end of url
    return (
        <div className="video-container">
            <div className='video'>
                <iframe
                    className='video-player' 
                    title={props.id}
                    // width={'560'} 
                    // height={'315'} 
                    src={embedUrl} 
                    frameBorder='0'
                    allow='autoplay; encrypted-media' 
                    allowFullScreen />
                    {/*iframe elements require unique title attributes*/}
            </div>
        </div>
    )
}