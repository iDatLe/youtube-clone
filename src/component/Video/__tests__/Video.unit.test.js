import React from 'react';
import { shallow } from 'enzyme';
import { Video } from '../Video';

test('renders correctly', () => {
    const wrapper = shallow(
        <Video id='-7fuHEEmEjs'/>
    )
    expect(wrapper).toMatchSnapshot();
});

test('renders null if there is no video inside of it', () => {
    const wrapper = shallow(
        <Video />
    )
    expect(wrapper).toMatchSnapshot();
});