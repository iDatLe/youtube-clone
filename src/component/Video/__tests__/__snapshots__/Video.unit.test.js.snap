// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      className="video-player"
      frameBorder="0"
      src="https://www.youtube.com/embed/-7fuHEEmEjs"
      title="-7fuHEEmEjs"
    />
  </div>
</div>
`;

exports[`renders null if there is no video inside of it 1`] = `""`;
